### PR TITLE
Distinguish float and integer values in JSON with decimal points

### PR DESCRIPTION
**Pull request by WKBae**
This commit ensures the distinction between float and integer values in JSON representation.
Previously, float values were not being stored correctly and were being recognized as integers. The addition of decimal points to these values, even if they are whole numbers, ensures that they are stored as the correct data type in JSON.

### CODE REVIEW

File: capnpc-go/capnpc-go.go [hunk 1]
```diff
@@ -916,12 +916,20 @@ func (t Type) json(w io.Writer) {
...
 		fprintf(w, "buf, err = json.Marshal(s);")
 		writeErrCheck(w)
 		fprintf(w, "_, err = b.Write(buf);")
 		writeErrCheck(w)
+	case TYPE_FLOAT32, TYPE_FLOAT64:
+		g_imported["encoding/json"] = true
+		g_imported["strconv"] = true
+		fprintf(w, "if float64(s) == float64(int64(s)) { buf = strconv.AppendFloat(buf[:0], float64(s), 'f', 1, 64) } else { buf, err = json.Marshal(s);")
+		writeErrCheck(w)
+		fprintf(w, "};")
```
Comment (Akais24) [inline capnpc-go/capnpc-go.go:930]: 이 친구는 왜 분리되었나요? 28번째 라인에 함께 하지 않고요

Comment (WKBae) [reply capnpc-go/capnpc-go.go:930]: writeErrCheck이 w에 `if err != nil { ... }`을 씁니다

Comment (Akais24) [reply capnpc-go/capnpc-go.go:930]: err가 함수 condition 밖에서 정의되어서
condition block 밖에서 검사해도 적당히 굴러가지 않을까요?

**Review by Akais24 — APPROVED** [review]

File: capnpc-go/capnpc-go.go (hunk 1 above)
Comment (WKBae) [reply capnpc-go/capnpc-go.go:930]: (구두 논의) 위쪽 if case에 불필요한 errCheck을 if-else 밖에 두기 그렇다

**Pull request merged**